src/guide/components/props.md

```diff
@@ -494,6 +494,39 @@ When prop validation fails, Vue will produce a console warning (if using the dev
 
 If using [Type-based props declarations](/api/sfc-script-setup.html#typescript-only-features), Vue will try its best to compile the type annotations into equivalent runtime prop declarations. For example, `defineProps<{ msg: string }>` will be compiled into `{ msg: { type: String, required: true }}`.
 
+
+<!-- added the equivalent validations for type based declarations -->
```

Comment:
**suggestion**: I don't think this belings to the `Guide`; probably it would be a nice addition to props API reference, would you mind moving it there?